### ISSUE TITLE
emoji autocompletion

### ISSUE THREAD

**Issue by tintou** [dino/dino#395]
It would be nice to be able to add an emoji by their namecode with colons:
: tada : becomes :tada: for instance

**Comment by muchweb** [comment]
Gnome already does this. Open context menu in any text input and select the 'Insert Emoji'.

Or install any of these programs: https://askubuntu.com/a/1045916

**Comment by Peque** [comment]
@muchweb I would say that does not resolve this issue. In my opinion it is much nicer to be able to use the `:emoji_namecode:` notation. :blush:

Other messaging software: Telegram-desktop, Riot, Slack... and web services: GitHub, Gitlab... use it too.

**Comment by muchweb** [comment]
Services that you have mentioned use plain text colon-style emoji codes that are rendered as images.

Dino makes use of Unicode emoji that are rendered with font of user's choice, which in my opinion is a better approach to this.

Maybe we can use 'rumoji' mapping to convert colon-style emojis into Unicode before message is sent to server.

**Comment by Peque** [comment]
@muchweb I agree that user's choice is a better approach. However, I can copy text from Telegram-desktop to GNOME Terminal and see the Emojis correctly displayed, so I am guessing it uses real Unicode emoji?

Also, those tools they do not only replace colon-style emojis into unicode before message is sent. They let you autocomplete with TAB and open a "pop-up" with suggested/recent emojis. Replacement occurs at that very same time, not like here in GitHub. That is part of the feature, and what I would consider most important (I would not care that much if emojis were rendered with font of user's choice or another).

**Comment by bobbybee** [comment]
A little late to the party, but I just implemented this as a Friday night "let's learn Vala" project (see linked MR) :)
